iohk-monitoring/src/Cardano/BM/Data/LogItem.lhs

```diff
@@ -111,10 +111,17 @@ instance FromJSON LOMeta where
 
 mkLOMeta :: MonadIO m => Severity -> PrivacyAnnotation -> m LOMeta
 mkLOMeta sev priv =
-    LOMeta <$> (liftIO getCurrentTime)
-           <*> (pack . show <$> (liftIO myThreadId))
+    LOMeta <$> liftIO getCurrentTime
+           <*> (cleantid <$> liftIO myThreadId)
            <*> pure sev
            <*> pure priv
+  where
+    cleantid threadid =
+        let stid = pack $ show threadid
+        in
+        if (isPrefixOf "ThreadId " stid)
```

Comment:
I understand the issue here - we want to strip off extraneous text - I've had a look at `base` package and I can't see any alternative interface.

However, the use of a '9' as a constant here worries me (I don't like constants in code other that '0' or '1'!).

Could I suggest using something like `stripPrefix` using a named constant e.g.
  `threadIdText = "ThreadId "
    condStripPrefix p s = maybe s id . stripPrefix p s
    clentId  = condStripPrefix threadIdText threadid . pack . show`

hope that makes sense - gets away from derived constants, makes the constant dependency clear